pkg/odo/cli/component/push.go

```diff
@@ -43,42 +45,73 @@ const PushRecommendedCommandName = "push"
 
 // PushOptions encapsulates options that push command uses
 type PushOptions struct {
-	ignores    []string
-	local      string
-	sourceType string
-	sourcePath string
-	*ComponentOptions
+	ignores          []string
+	local            string
+	sourceType       string
+	sourcePath       string
+	localConfig      *config.LocalConfigInfo
+	componentContext string
+	*genericclioptions.Context
+	client *occlient.Client
 }
 
 // NewPushOptions returns new instance of PushOptions
 func NewPushOptions() *PushOptions {
-	return &PushOptions{[]string{}, "", "", "", &ComponentOptions{}}
+	return &PushOptions{
+		ignores:     []string{},
+		localConfig: &config.LocalConfigInfo{},
+	}
 }
 
 // Complete completes push args
 func (po *PushOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
-	err = po.ComponentOptions.Complete(name, cmd, args)
-	if err != nil {
-		return err
+	po.client = genericclioptions.Client(cmd)
+
+	if len(po.componentContext) > 0 {
+		err = po.localConfig.GetLocalConfigFileFromPath(po.componentContext)
+		if err != nil {
+			return errors.Wrapf(err, "failed to access component config from %+v", po.componentContext)
+		}
+
+		err = po.localConfig.Loadconfig()
+		if err != nil {
+			return errors.Wrapf(err, "failed to load compoennt settings from %s", po.componentContext)
+		}
+	} else {
+		conf, err := config.NewLocalConfigInfo()
+		if err != nil {
+			return errors.Wrap(err, "failed to fetch component config")
+		}
+		po.localConfig = conf
 	}
 
-	po.sourceType, po.sourcePath, err = component.GetComponentSource(po.Context.Client, po.componentName, po.Context.Application)
-	if err != nil {
-		return errors.Wrapf(err, "unable to get component source")
+	if _, err = os.Stat(po.localConfig.Filename); err != nil {
+		return errors.Wrapf(err, "failed trying to read config file in %s", po.localConfig.Filename)
```

Comment:
See above comment, we should implement this too into a separate function: "LoadConfig()" or something.

Reply:
fixed it now as in above comment